workitem/workitemtype_repository.go

```diff
@@ -75,6 +76,28 @@ func (r *GormWorkItemTypeRepository) Load(ctx context.Context, spaceID uuid.UUID
 	return &res, nil
 }
 
+// Exists returns true if a work item type exists with a given ID
+func (m *GormWorkItemTypeRepository) Exists(ctx context.Context, spaceID uuid.UUID, id uuid.UUID) (bool, error) {
```

Comment:
This function doesn't respect the cache that we have for WITs. Also, why does this function have a spaceID as a parameter? Shouldn't the WIT ID be sufficient? In fact, in the future a WIT is no longer bound to a space, so I suggest to remove the spaceID from here.

In my branch, I've already implemented this function like so:

```go

// Exists returns true if the work item with the given ID exists
func (r *GormWorkItemTypeRepository) Exists(ctx context.Context, id uuid.UUID) (bool, error) {
	log.Info(ctx, map[string]interface{}{
		"wit_id": id,
	}, "Checking if work item type exists")
	_, exists := cache.Get(id)
	if exists {
		return true, nil
	}
	log.Info(ctx, map[string]interface{}{
		"wit_id": id,
	}, "Work item type doesn't exist in the cache. Checking in DB...")
	query := fmt.Sprintf("SELECT EXISTS(SELECT id FROM %s WHERE id=$1 AND deleted_at IS NULL)", WorkItemType{}.TableName())
	err := r.db.CommonDB().QueryRow(query, id.String()).Scan(&exists)
	if err != nil {
		log.Error(ctx, map[string]interface{}{
			"wit_id": id,
			"err":    err,
		}, "failed to check if work item type exists")
		return false, errs.Wrap(err, "failed to check if work item type exists")
	}
	return exists, nil
}
```